src/core/server/saved_objects/permission_control/client.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { OpenSearchDashboardsRequest } from '../../http';
+import { SavedObjectsServiceStart } from '../saved_objects_service';
+import { SavedObjectsBulkGetObject } from '../service';
+
+export enum PermissionMode {
```

Comment:
Could we move this `PermissionMode` to some common folder? So we can use it even in `public` folder.